### PR TITLE
Feature/nbpillai/renter candidate page

### LINKED ISSUE #75: Renter Candidate Page: Details Section
# Overview

Implement just the details section of the Renter Candidate page.

## Frontend

Implement the top half of the Renter Candidate page. You can put the page at the route `/candidate/{id}`.

[Figma](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=2787%3A14562&mode=dev)

image

## Backend

Implement the `GET /renter-candidates/:id` route ([reference](https://docs.google.com/document/d/19EBAxzbCyPdT6RgELrGsq6A2RPeDt2OSi5DYbFWm03k/edit#bookmark=id.dho87e3k6jj4)).

Make sure to put business logic (e.g. functions that query the database) in the `backend/src/services/renter.ts` file.

On the frontend, call this route with the `{id}` path param to fetch the data for the RC.

## Acceptance Criteria
- [ ] Going to `http://localhost:5173/candidate/65e369c72751c66790e71f89` (or any other valid RC id) properly displays the information for the RC

### PR DESCRIPTION
**Pull request by nbpillai**
## Tracking Info

Resolves #75 

## Changes

I implemented the frontend of the Renter Candidate page and a get route on the backend to get the details of a renter candidate and display them on the page. 

## Testing

I tested it on sample renter candidate and the information was displayed correctly. I also tested the get route on postman.

## Confirmation of Change
Screen Shot 2024-04-16 at 12 54 21 PM

### CODE REVIEW

File: frontend/src/pages/RenterCandidate.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+  const InfoText = styled(InfoHeader)`
+    font-weight: 400;
+  `;
+
+  const EditButton = styled(Button)`
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+  `;
```
Comment (petabite) [inline frontend/src/pages/RenterCandidate.tsx]: Can we move these outside of the function for [performance reasons](https://styled-components.com/docs/faqs#why-should-i-avoid-declaring-styled-components-in-the-render-method)?

**Review by petabite — CHANGES_REQUESTED** [review]
Looks good! Left a small comment below. Can you also fix the lint issues?

File: frontend/src/pages/RenterCandidate.tsx (hunk 1 above)
Comment (nbpillai) [reply frontend/src/pages/RenterCandidate.tsx]: fixed!

**Review by YashRavipati1 — APPROVED** [review]
LGTM!

**Review by petabite — APPROVED** [review]
LGTM 🚢

**Pull request merged**